src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,80 @@
+package nextstep.subway.application;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Station;
+import nextstep.subway.domain.StationRepository;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+    private LineRepository lineRepository;
+    private StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(final LineRequest lineRequest) {
+        Station upStation = stationRepository.findById(lineRequest.getUpStationId()).get();
+        Station downStation = stationRepository.findById(lineRequest.getDownStationId()).get();
+        return LineResponse.of(lineRepository.save(lineRequest.toLine(upStation, downStation)));
+    }
+
+    public List<LineResponse> findAllLine() {
+        List<Line> lines = lineRepository.findAll();
+        return lines.stream().map(LineResponse::of).collect(Collectors.toList());
+    }
+
+    public LineResponse findLine(final Long id) {
+        return LineResponse.of(lineRepository.findById(id).get());
```

Comment:
조회한 id의 노선이 없을 경우 예외처리를 해줄 수 있겠네요 :)
```suggestion

    public LineResponse findLine(final Long id) {
        Line line = lineRepository.findById(id).orElseThrow(EntityNotFoundException::new);
        return LineResponse.of(line);
```

Reply:
네 조언해 주신것 처럼 Exception 으로 처리하는게 가독성 측면에서도 좋은것 같습니다 ^^
감사합니다~